src/gnome-mastermind.c

```diff
@@ -2035,162 +1930,77 @@ static void preferences_action (void){
-    N_("_Quit"), "<control>Q", 
-    N_("Quit the game"),
-    G_CALLBACK (quit_action) },
-
-  { "SettingsMenuAction", NULL, N_("_Settings") },
-
-  { "PrefAction", GTK_STOCK_PREFERENCES,
-    N_("_Preferences"), "<control>P", 
-    N_("Change game settings"),
-    G_CALLBACK (preferences_action) },
-
-  { "HelpMenuAction", NULL, N_("_Help") },
- 
-  { "AboutAction", GTK_STOCK_ABOUT,
-    N_("_About"), 
-    NULL,
-    N_("About"),
-    G_CALLBACK (about_action) },
-
-  { "HelpAction", GTK_STOCK_HELP, 
-    N_("_Contents"),
-    "F1",
-    N_("_Contents"),
-    G_CALLBACK (help_action) }
-};
-
-static void show_tb_callback (void) {
+static void show_tb_callback (GSimpleAction *action, GVariant *param,
+                              gpointer data) {
   gboolean state;
 
-  state =
-    gtk_toggle_action_get_active (GTK_TOGGLE_ACTION (show_toolbar_action));
+  state = g_variant_get_boolean (param);
 	
   gm_debug ("gc_show_toolbar: %d state: %d\n", gc_show_toolbar, state);
  
-  gconf_client_set_bool (settings, "/apps/gnome-mastermind/show_toolbar", state , NULL);
+  g_settings_set_boolean (settings, "show-toolbar", state);
+  g_simple_action_set_state (action, param);
 }
 
 
-static const GtkToggleActionEntry toggle_actions[] = {
-  {"ShowToolbarAction", NULL, N_("_Show Toolbar"), NULL, N_("Show or hide the toolbar"),
-   G_CALLBACK (show_tb_callback)}
+/* Create a list of entries which are passed to the Action constructor.
+ * This is a huge convenience over building Actions by hand. */
+static GActionEntry entries[] =
+{
+	{ "new", new_game, NULL, NULL, NULL },
+	{ "undo", undo_action_cb, NULL, NULL, NULL },
+	{ "redo", redo_action_cb, NULL, NULL, NULL },
+	{ "quit", quit_action, NULL, NULL, NULL },
+	{ "show-toolbar", NULL, NULL, "true", show_tb_callback },
+	{ "prefs", preferences_action, NULL, NULL, NULL },
+	{ "help", help_action, NULL, NULL, NULL },
+	{ "about", about_action, NULL, NULL, NULL }
 };
 
 static guint n_entries = G_N_ELEMENTS (entries);
 
-/* Implement a handler for GtkUIManager's "add_widget" signal. The UI manager
- * will emit this signal whenever it needs you to place a new widget it has. */
 static void
-menu_add_widget (GtkUIManager *ui, GtkWidget *widget, GtkContainer *container)
-{
-  gtk_box_pack_start (GTK_BOX (container), widget, FALSE, FALSE, 0);
-  gtk_widget_show (widget);
-}
-
-int main ( int argc, char *argv[] )
+activate (GApplication *app)
 {
   int i;
 
   GtkWidget *gridframe;
-  GtkActionGroup *action_group; 
-  GtkUIManager *menu_manager; 
-  GtkAccelGroup *accel_group;
-  const gchar *debug_env = NULL;
+  GtkBuilder *builder;
 
-  bindtextdomain (GETTEXT_PACKAGE, GMLOCALEDIR);
-  bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
-  textdomain (GETTEXT_PACKAGE); 
-
-  debug_env = g_getenv ("GM_DEBUG");
-  if (debug_env && !g_ascii_strcasecmp(debug_env, "yes")) {
-    gm_debug_on = TRUE; 
-    gm_debug("*** DEBUG MODE ON ***\n");
-  } else {
-    gm_debug_on = FALSE;
-  }
```

Comment:
not sure why this was removed while keeping the debug calls elsewhere, doesn't seems something really related to the gtk3 port anyway

Reply:
ignore this, I didn't notice it was moved in a dedicated function :-D